### DISCUSSION TITLE
Diedrich IR-2.5 Gas and Air Not Reading After First Roast in a Session

### DISCUSSION THREAD

**Discussion by HilloBeans** [artisan-roaster-scope/artisan#687] [Q&A]
Hi All,

My first post here . . .

I have a new IR-2.5 and I've done about 40 batches using Artisan.  I get readouts for all my sensors on the first batch of a session, then the gas and air stop reading (both the LED and the graph).  I've tried quitting and restarting the software and unplugging/re-plugging the USB, but nothing changes.  Diedrich says they can't help with this one. I am on a Mac (not plugged in to power).

Can anyone offer a suggestion?

Many thanks!!

**Comment by MAKOMO** [comment]
Dear HilloBeans,

could you please attach here your artisan-settings.aset file (export via menu Help >> Save Settings, rename from .aset into .txt and then drag and drop here on GitHub to the answer text message section. I will take a look into this. 

Does this happen on each of those 40 batches? If restart Artisan and unplugging does not help, how did you get that data logged on the next session? Does restarting the Mac resolved this?

**Comment by HilloBeans** [comment]
Thanks for your quick reply, Marko.  File its attached here.  I usually roast 5-6 batches a session, and this pattern is consistent with the first batch of each session having the problem. For the remaining batches in a session, all other sensors display properly.  I have not tried restarting the computer after the first batch, but I will do that when I roast next.

[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/7178312/artisan-settings.txt)

**Comment by MAKOMO** [comment]
Wait a moment. Before you stated that "I get readouts for all my sensors on the first batch of a session , then the gas and air stop reading" now you state "the first batch of each session having the problem. For the remaining batches in a session, all other sensors display properly". 

Is the first batch of a session now trouble free, or having the problem (which I deduce is to not show gas and air readings)? I am confused.

**Comment by HilloBeans** [comment]
Apologies, I didn’t say that right. The first batch of a session has no
problems and the subsequent batches do.

**Comment by MAKOMO** [comment]
I don't see anything wrong with this settings. In a first step I would ensure to have the latest Phidget driver installed and all modules running on the latest firmware. Install/update the Phidget driver and then open the Phidget Control Panel app and check for firmware updates (see example window below). For modules with names in red a firmware update is available. Click this name and update the modules firmware. Test again if this improves the stability. If this does not help, use a different or shorter USB cable. If this does not help, there might be something wrong with the Phidget IO module (I guess Dietrich is still using the Phidget 1018),

<img width="754" alt="Screenshot 2021-09-16 at 15 52 04" src="https://user-images.githubusercontent.com/13909011/133671177-ddfed49a-c795-4b0d-a553-4a6d75a74031.png">

**Comment by HilloBeans** [comment]
Marko, circling back on this.  There was a Phidget update for the Mac
September 21, and that didn't change anything.  I've bought two shorter
cables and that didn't help - and DIedrich passses the buck back to Artisan.

The performance is still pretty erratic.  The gas and air readings now
rarely show on the LCDs at all, and I've changed my Axes to show only
temperature results.  I'll just roll with this for now, as my gas and air
changes are pretty consistent.

I need to send you guys a donation, and I'll do that soon.

All the best,

Kent

**Reply by MAKOMO** [reply]
The code on the Artisan side is thin and no problems have been reported since years with this, and there are quite some Artisan users running Phidgets. Most likely this is a hardware issue and the hardware was sold by Dietrich to you, right? Most of the software here comes from Phidgets as Artisan is just calling getData(). Not much one can make wrong there. Sorry to hear that those issues did not get resolved. You could check what the Phidget Control Panel is showing for those gas and air readings. I could bet that they are missing there too as the sensor or device which is sending the data is erratic.

**Comment by HilloBeans** [comment]
I'll check that - thanks so much for this advice.  I am not super stoked
with the Diedrich support so far.  I'll stay in touch and let you know what
I learn.
